src/object/C4Movement.cpp

```diff
@@ -658,11 +841,17 @@ bool SimFlightHitsLiquid(C4Real fcx, C4Real fcy, C4Real xdir, C4Real ydir)
 	// Start in water?
 	int temp;
 	if (DensityLiquid(GBackDensity(fixtoi(fcx), fixtoi(fcy))))
-		if (!SimFlight(fcx, fcy, xdir, ydir, 0, C4M_Liquid - 1, temp=10))
+	{
+		if (!SimFlight(fcx, fcy, xdir, ydir, 0, C4M_Liquid - 1, temp = 10))
```

Comment:
With the introduction of that block, `temp` can be moved inside and initialized there instead of that horrible assignment during parameter passing. (Same below.)

Reply:
This is resolved now, right?